task/tekton-catalog-publish/0.1/tekton-catalog-publish.yaml

```diff
@@ -0,0 +1,69 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: tekton-catalog-publish
+  labels:
+    app.kubernetes.io/version: "0.1"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.17.0"
+    tekton.dev/tags: catalog, bundles
+    tekton.dev/displayName: "Publish a Tekton Catalog"
+spec:
+  description: >-
+    This task publishes a Tekton catalog as bundles to a container registry
+  workspaces:
+    - name: catalog
+      description: >-
+        A workspace that holds the catalog to be published. The catalog must be stored
+        in the root of the workspace mount point, and is must follow the
+        [Tetkon Catalog](https://github.com/tektoncd/catalog#catalog-structure) structure.
+    - name: dockerconfig
+      description: >-
+        An optional workspace that allows providing a .docker/config.json
+        file for tkn to access the container registry. The file should be placed at
+        the root of the Workspace with name config.json.
+      optional: true
+
+  params:
+    - name: TKN_IMAGE
+      type: string
+      description: tkn CLI container image to run this task
+      default: gcr.io/tekton-releases/dogfooding/tkn@sha256:015bb6018d5eaf4523a775f4955c85f68077099753661941d06843d5a61a86df
+    - name: REGISTRY
+      type: string
+      description: The registry where bundles are published to
+    - name: PATH
+      type: string
+      description: The path in the registry
+      default: tekton/catalog/tasks
+    - name: TAG
+      type: string
+      description: An optional extra tag. If provided, tasks are tagged with it too.
+      default: ""
+
+  steps:
+    - name: publish
+      image: "$(params.TKN_IMAGE)"
+      workingDir: "$(workspaces.catalog.path)"
+      env:
+        - name: REGISTRY
+          value: $(params.REGISTRY)
+        - name: REGISTRY_PATH
+          value: $(params.PATH)
+        - name: TAG
+          value: $(params.TAG)
+      script: |
+        set -e
+        TARGET="${REGISTRY}"
+        [[ "${REGISTRY_PATH}" != "" ]] && TARGET="${TARGET}/${REGISTRY_PATH}"
+        cd task
+        find . -type f -mindepth 3 -maxdepth 3 -name '*yaml' | while read aa; do
+          FOLDER=$(dirname $aa)
+          VERSION=$(basename $FOLDER)
+          TASK=$(basename $(dirname $FOLDER))
+          cat $aa | tkn bundle push ${TARGET}/${TASK}:${VERSION} -f- &
```

Comment:
Both the `tkn bundle push` are placed in the background; `&` — and then the task _sleeps_ for 0.1 seconds.  But it feels more correct to just run the tkn bundle push command without putting it in the background.

* If the push takes more than 0.1 second, you risk the processes be killed by k8s as the pod is being torn down
* Putting the tkn bundle push in the background, causes any error to be hidden.  The task exit code will be 0 because they are in the background.

I'm just wondering if there is a reason for putting them in the background that I wasn't aware of.

Reply:
Good point, thank you! Removed the `&`